tests/infer/test_abstract_infer.py

```diff
@@ -28,7 +36,49 @@ def test_posterior_predictive():
     mcmc_run = MCMC(nuts_kernel, num_samples=1000, warmup_steps=200).run(num_trials)
     posterior_predictive = TracePredictive(model, mcmc_run, num_samples=10000).run(num_trials)
     marginal_return_vals = posterior_predictive.marginal().empirical["_RETURN"]
-    assert_equal(marginal_return_vals.mean, torch.ones(5) * 700, prec=30)
+    assert_close(marginal_return_vals.mean, torch.ones(5) * 700, rtol=0.05)
+
+
+def test_posterior_predictive_svi_manual_guide():
+    true_probs = torch.ones(5) * 0.7
+    num_trials = torch.ones(5) * 1000
+    num_success = dist.Binomial(num_trials, true_probs).sample()
+    conditioned_model = poutine.condition(model, data={"obs": num_success})
+    opt = optim.Adam(dict(lr=1.0))
+    loss = Trace_ELBO()
+    guide = beta_guide
+    svi_run = SVI(conditioned_model, guide, opt, loss, num_steps=1000, num_samples=100).run(num_trials)
+    posterior_predictive = TracePredictive(model, svi_run, num_samples=10000).run(num_trials)
```

Comment:
Let us run this using `.run(num_trials[:3])` and change the assertion below to `assert_close(marginal_return_vals.mean, torch.ones(3) * 700, rtol=0.05)`, to make sure that we are addressing the second case correctly.